Tests/integration/Ping.Tests.ps1

```diff
@@ -0,0 +1,55 @@
+﻿#
+# Copyright 2018, Alexis La Goutte <alexis dot lagoutte at gmail dot com>
+# Copyright 2018, Cédric Moreau <moreaucedric0 at gmail dot com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+. ../common.ps1
+
+Describe  "Ping (ipv4_address)" {
+    beforeAll {
+        $script:ipv4_gateway = (Get-ArubaSWSystem).default_gateway.octets
+    }
+
+    It "Ping a valid IP(v4) address (Default Gateway:$script:ipv4_gateway)" {
+        $ping = Test-ArubaSWPing -ipv4_address $script:ipv4_gateway
+        $ping.result | Should -Be "PR_OK"
+        $ping.rtt_in_milliseconds | Should -Not -BeNullOrEmpty
+    }
+
+    It "Ping a invalid IP(v4) address (0.0.0.0)" {
+        $ping = Test-ArubaSWPing -ipv4_address 0.0.0.0
+        $ping.result | Should -Be "PR_INVALID_ADDRESS"
+    }
+}
+
+
+Describe  "Ping (-hostname)" {
+
+    BeforeAll {
+        #Always remove DNS Settings...
+        Remove-ArubaSWDns -noconfirm
+    }
+
+    It "Ping a hostname (without DNS config)" {
+        $ping = Test-ArubaSWPing -hostname www.arubanetworks.com
+        $ping.result | Should -Be "PR_UNABLE_TO_RESOLVE_HOST_NAME"
+    }
+
+    It "Ping a hostname (with DNS config $pester_dns1 / $pester_dns2)" {
+        #Configure DNS
+        Set-ArubaSWDns -mode Manual -server1 $pester_dns1 -server2 $pester_dns2
+        $ping = Test-ArubaSWPing -hostname www.arubanetworks.com
+        $ping.result | Should -Be "PR_OK"
+        $ping.rtt_in_milliseconds | Should -Not -BeNullOrEmpty
+    }
+
+    AfterAll {
+        #Always remove DNS Settings...
+        Remove-ArubaSWDns -noconfirm
```

Comment:
What about previous DNS servers configured before the test ? Don't they need to be re-added ?

Reply:
yes... but it is not already the case on DNS test...